### ISSUE TITLE
Should allow None as initial value when use type comments

### ISSUE THREAD

**Issue by NateScarlet** [microsoft/pyright#1649]
**Describe the bug**
[PEP484](https://www.python.org/dev/peps/pep-0484/#type-comments) says None should allowed as intial value.

> Type checkers should not complain about this (despite the value None not matching the given type), nor should they change the inferred type to Optional[...] (despite the rule that does this for annotated arguments with a default value of None). The assumption here is that other code will ensure that the variable is given a value of the proper type, and all uses can assume that the variable has the given type.

**To Reproduce**
See code below

**Expected behavior**
Allow use `None` as initial value

**Screenshots or Code**
```python
from typing import IO

stream = None  # type: IO[str]
```
![image](https://user-images.githubusercontent.com/21083014/111602625-7f348480-880e-11eb-8eb4-f4bc3952a4d4.png)

**VS Code extension or command-line**
Pylance: 2021.3.2

**Comment by erictraut** [comment]
This is an intentional choice for type safety. If the variable is set to `None`, but its type is declared such that it does not permit `None`, then a bunch of potential bugs will go undetected.

**Comment by NateScarlet** [comment]
If i read the PEP correctly, this is a alternative syntax of `stream: IO[str]` for python3.5 and earlier. 

I am ok if it will not get supported since i can use `stream = cast(IO[str], None)` instead.

**Comment by erictraut** [comment]
Using a cast is also unsafe. I recommend:
```python
stream: Optional[IO[str]] = None
```

Then when you use `self.stream`, you should verify that it has been initialized to a non-None value.
